File: pdu/python/hako_srv_msgs/pdu_cdr_conv_DetachResponse.py
from .pdu_pytype_DetachResponse import DetachResponse
from ..pdu_cdr_runtime import CdrReader, CdrWriter




def py_to_cdr_body_DetachResponse(writer: CdrWriter, src: DetachResponse):
    writer.write_uint32(src.result_code)


def cdr_body_to_py_DetachResponse(reader: CdrReader, dst: DetachResponse):
    dst.result_code = reader.read_uint32()
    return dst


def py_to_cdr_DetachResponse(src: DetachResponse) -> bytes:
    writer = CdrWriter()
    writer.write_encapsulation()
    py_to_cdr_body_DetachResponse(writer, src)
    return writer.bytes()


def cdr_to_py_DetachResponse(cdr_payload) -> DetachResponse:
    reader = CdrReader(cdr_payload)
    reader.read_encapsulation()
    dst = DetachResponse()
    return cdr_body_to_py_DetachResponse(reader, dst)

File: pdu/javascript/hako_srv_msgs/pdu_cdr_conv_DetachResponse.js
import { DetachResponse } from './pdu_jstype_DetachResponse.js';
import { PduCdrWriter, PduCdrReader } from '../pdu_cdr_runtime.js';


export class PduDetachResponseConverter {
    /**
     * @param {PduCdrWriter} writer
     * @param { DetachResponse } src
     */
    static to_cdr_body(writer, src) {
        writer.write_uint32(src.result_code);
    }

    /**
     * @param {PduCdrReader} reader
     * @param { DetachResponse } dst
     * @returns { DetachResponse }
     */
    static cdr_body_to_js(reader, dst) {
        dst.result_code = reader.read_uint32();
        return dst;
    }

    /**
     * @param { DetachResponse } src
     * @returns {ArrayBuffer}
     */
    static to_cdr(src) {
        const writer = new PduCdrWriter();
        writer.write_encapsulation();
        this.to_cdr_body(writer, src);
        return writer.get_buf();
    }

    /**
     * @param {ArrayBuffer|ArrayBufferView} cdrPayload
     * @returns { DetachResponse }
     */
    static from_cdr(cdrPayload) {
        const reader = new PduCdrReader(cdrPayload);
        reader.read_encapsulation();
        return this.cdr_body_to_js(reader, new DetachResponse());
    }
}
